### PR TITLE
ocamlPackages.elpi: 1.11.2 -> 1.11.4

### PR DESCRIPTION
**Pull request by CohenCyril**
and fixing dependencies

###### Motivation for this change

Upgrading elpi and coq-elpi

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build ocamlPackages.elpi coqPackages_8_11.coq-elpi coqPackages_8_12.coq-elpi coqPackages_8_11.hierarchy-builder coqPackages_8_12.hierarchy-builder

File: pkgs/development/ocaml-modules/elpi/default.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+     sha256 = "1hmjp2z52j17vwhhdkj45n9jx11jxkdg2dwa0n04yyw0qqy4m7c1";
    };
 
   minimumOCamlVersion = "4.04";
 
-  buildInputs = [ ppx_tools_versioned ];
+  buildInputs = [ perl ppxlib ];
 
-  propagatedBuildInputs = [ camlp5 ppx_deriving re ];
+  propagatedBuildInputs = [ ncurses camlp5 ppx_deriving re ];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/elpi/default.nix:18]: Why this change?

Comment (CohenCyril) [reply pkgs/development/ocaml-modules/elpi/default.nix:18]: According to @gares if I understood well `elpi` may need `tput` at runtime to display outputs in some context, and `ncurses` was the most sensible derivation exposing it... would some other changes make more sense?

Comment (vbgl) [reply pkgs/development/ocaml-modules/elpi/default.nix:18]: propagatedBuildInputs is not for runtime dependencies. You may either wrap the elpi binaries to set the PATH environment variable or hardcode the path to tput in the source code.

Comment (CohenCyril) [reply pkgs/development/ocaml-modules/elpi/default.nix:18]: @vbgl thanks! I do not know how to implement either of these solutions, can you give me some doc and/or examples?

Comment (vbgl) [reply pkgs/development/ocaml-modules/elpi/default.nix:18]: For instance, the opam binary is wrapped: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/ocaml/opam/default.nix#L104

Coq hardcodes the path to csdp: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/logic/coq/default.nix#L44

A third possibility is just to leave a comment: “elpi will assume your terminal is 80 positions wide, unless tput is found in PATH”.

Comment (CohenCyril) [reply pkgs/development/ocaml-modules/elpi/default.nix]: @vbgl, thanks a lot for you advice, I took the option to hardcode the path and it works well, thank you!

**Comment by CohenCyril** [comment]
@GrahamcOfBorg build ocamlPackages.elpi coqPackages_8_11.coq-elpi coqPackages_8_12.coq-elpi coqPackages_8_11.hierarchy-builder coqPackages_8_12.hierarchy-builder

**Comment by vbgl** [comment]
@GrahamcOfBorg build coqPackages_8_11.hierarchy-builder coqPackages_8_12.hierarchy-builder

File: pkgs/development/ocaml-modules/elpi/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/elpi/default.nix]: You’re welcome. Thanks for your contributions.

**Comment by CohenCyril** [comment]
Thanks!
